src/signatures/ed25519.rs

```diff
@@ -148,3 +158,21 @@ impl Signature {
         Self(ed25519_zebra::Signature::from(bs))
     }
 }
+
+impl PartialOrd for Signature {
```

Comment:
Why do we need to compare signatures?

Reply:
Addressed above. We use it as part of UTXO.

Reply:
Alright. I thought that signature and public key **bytes** (not types) are compared/ordered as part of UTxO. Corresponding types should probably respect order in binary representation.

Reply:
Yes, we use the bytes. I had hoped that we could use the representation from crypto to avoid converting back-and-forth, but if you think that's unwise then I will abort.

Reply:
Has the comment been addressed?

Reply:
Yes

Reply:
The current implementation via byte array is correct. It's possible to access inner signature fields r and s to avoid copying. Also, order on Signature type is total. I'd do it like this:

```rust
impl PartialOrd for Signature {
    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
        Some(self.cmp(other))
    }
}

impl Ord for Signature {
    fn cmp(&self, other: &Self) -> Ordering {
        let r_cmp = self.0.r_bytes().cmp(other.0.r_bytes());
        let s_cmp = self.0.s_bytes().cmp(other.0.s_bytes());
        r_cmp.then(s_cmp)
    }
}
```

To enable access to r and s bytes we need to update ed25519-zebra to version "4.0".

Reply:
It would be nice to ensure/test property `for all s1, s2: [u8; 64] => s1.cmp(s2) == Signature::from_bytes(s1).cmp(&Signature.from_bytes(s2))`. This should hold as ed25519_zebra-v4.0 uses [ed25519::Signature](https://github.com/RustCrypto/signatures/blob/master/ed25519/src/lib.rs#L303) which keeps r and s components as bytes without and normalization.

Reply:
Signature cmp uses internal r/s values to avoid copying. Signature cmp respects corresponding byte ordering. Added tests.